bench/wtperf/wtperf.c

```diff
@@ -2308,6 +2313,31 @@ main(int argc, char *argv[])
 	if ((ret = config_sanity(cfg)) != 0)
 		goto err;
 
+	/* Backup the config */
+	req_len = strlen(cfg->home) + 100;
+	if ((path = calloc(req_len, 1)) == NULL) {
+		(void)enomem(cfg);
+		goto err;
+	}
+
+	snprintf(path, req_len + 14, "%s/CONFIGBACKUP", cfg->home);
+	if ((fp = fopen(path, "w")) == NULL) {
+		lprintf(cfg, errno, 0, "%s", path);
+		goto err;
+	}
+
+	/* Print the config dump */
+	fp = fopen(path, "w+");
+	while (!TAILQ_EMPTY(&cfg->config_head)) {
+		config_line = TAILQ_FIRST(&cfg->config_head);
+		TAILQ_REMOVE(&cfg->config_head, config_line, c);
+		fprintf(fp, "%s\n", config_line->string);
+		free(config_line->string);
+		free(config_line);
+	}
+	free(path);
+	fclose(fp);
```

Comment:
I'd prefer if all of this code was in its own function, {{config_print}} and called, similar to {{config_sanity()}} rather than mainlined in {{main()}}.  I suggest {{config_print}} because that is the name of the function used in test/format.